### DISCUSSION TITLE
Running nvim-lint on InsertLeave does not happen for phpstan

### DISCUSSION THREAD

**Discussion by kauffinger** [LazyVim/LazyVim#1785] [Q&A]
I am trying to migrate from none-ls. But, when i try to run phpstan on InsertLeave with nvim-lint it only runs when opening or writing a buffer.
```lua
return {
  "mfussenegger/nvim-lint",
  event = "LazyFile",
  opts = {
    -- Event to trigger linters
    events = { "BufWritePost", "BufReadPost", "InsertLeave" },
    linters_by_ft = {
      fish = { "fish" },
      php = { "phpstan" },
    },
    ---@type table<string,table>
    linters = {
      -- -- Example of using selene only when a selene.toml file is present
      -- selene = {
      --   -- `condition` is another LazyVim extension that allows you to
      --   -- dynamically enable/disable linters based on the context.
      --   condition = function(ctx)
      --     return vim.fs.find({ "selene.toml" }, { path = ctx.filename, upward = true })[1]
      --   end,
      -- },
    },
  },
}
```
I think I can see the registered autocommand:
```
nvim-lint  InsertLeave
*<Lua 1059: ~/.local/share/nvim/lazy/LazyVim/lua/lazyvim/plugins/linting.lua:46>
```

> If I remove `InsertLeave` from the `events` in `opts`, I cannot see the autocommand registered for it. So that seems to work. I have no idea why it could still be the case that it won't run on that event.
> 
Upon further testing, it seems other linters (hadolint, pylint) work fine on InsertLeave. I am lost for ideas why phpstan does not do that.

Here is my full current config for reference, if anyone finds the time: [Repo](https://github.com/kauffinger/lazyvim/tree/migration-away-from-none-ls)
[nvim-lint config](https://github.com/kauffinger/lazyvim/blob/migration-away-from-none-ls/lua/plugins/nvim-lint.lua)

I am sorry if this is obvious, I am not that experienced. Thank you for your time!

**Comment by kauffinger** [comment]
Nevermind, could this be due to phpstan running on files, thus needing a written buffer to see changes?

If that is the case, a none-ls equivalent config would need some additional workarounds, am I correct?

**Comment by folke** [comment]
That's proythe reason indeed. Either way, not a lazyvim issue. Closing

**Reply by kauffinger** [reply]
Thanks for the feedback, I appreciate your efforts! Sorry for the hassle!
